### PR TITLE
#2367 Show `move to inbox` button for all cases when it's not inbox

### LINKED ISSUE #2367: mail in All Mail (archived) cannot be moved back to inbox
most can, but sometimes it can't, missing icon

example here

image

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR shows `move to inbox` button for all cases when it's not inbox

close #2367 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by sosnovsky — COMMENTED** [review]
Looks good, but I noticed a small issue:
- compose and send new email
- go to Sent folder (or All Mail), open sent message
- toolbar shows `move to inbox` button, but should show `move to archive` button (as Gmail app does)

**Comment by ioanmo226** [comment]
I believe Gmail's logic for handling sent emails is flawed. 
Archiving `Sent emails/threads` doesn't seem to have a purpose.(it actually does nothing)

Instead, offering a 'Move to Inbox' option for sent items would make more sense to me

**Comment by ioanmo226** [comment]
Yep!. Nice catch. let me fix it

File: FlowCrypt/Controllers/Inbox/InboxViewController+TableView.swift [hunk 1]
```diff
@@ -417,7 +417,7 @@ extension InboxViewController {
         case .moveToTrash, .permanentlyDelete:
             removeMessage(at: indexToUpdate)
         case .archive, .moveToInbox:
-            if path.isEmpty { // no need to remove in 'All Mail' folder
+            if path.isEmpty || path == "SENT" { // no need to remove in 'All Mail' folder
```
Comment (sosnovsky) [inline FlowCrypt/Controllers/Inbox/InboxViewController+TableView.swift:420]: it looks like we should remove messages only on `Inbox` screen, as moving to inbox or archiving doesn't remove messages from other folders.

For example, go to some label folder, archive message there - it should still stay in messages list of this label, but currently it's removed

Comment (ioanmo226) [reply FlowCrypt/Controllers/Inbox/InboxViewController+TableView.swift:420]: Yep! Correct

**Review by sosnovsky — APPROVED** [review]
Works well now 👍

**Pull request merged**